### PR TITLE
Feature/timi/android getting started

### PR DESCRIPTION
**Pull request by timMikkelsen**

### CODE REVIEW

File: src/android/v3/getting-started/directions.md [hunk 1]
```diff
@@ -14,9 +14,374 @@ eleventyNavigation:
 <!-- Directions -->
 {% include "src/shared/getting-started/directions/directions.md" %}
 
-* **TODO: How to get directions between two coordinates and render it on the map (MPDirectionsService, MPDirectionsQuery, MPRoutingProvider, onResultListener, MPDirectionsRenderer)**
+After having created our list of search results we have a good starting point to creating directions between two locations.
```
Comment (marksdk) [inline src/android/v3/getting-started/directions.md:17]:
```suggestion
After having created our list of search results, we have a good starting point for creating directions between two Locations.
```

File: src/android/v3/getting-started/directions.md [hunk 2]
```diff
@@ -14,9 +14,374 @@ eleventyNavigation:
...
+Since our search only supports a single search we will hardcode a location coordinate into our app and use that as the basis for our origin, we'll then create a route and navigate to a view of the the navigation details and show the route on the map from our origin point to the destination.
```
Comment (marksdk) [inline src/android/v3/getting-started/directions.md:18]:
```suggestion
Since our search only supports a single search, we will hardcode a Location's coordinate into our app, and use that as the basis for our Origin. Then we'll create a route, navigate to a view of the navigation details, and show a route on the map from the Origin to the Destination.
```

File: src/android/v3/getting-started/directions.md [hunk 3]
```diff
@@ -14,9 +14,374 @@ eleventyNavigation:
...
+
+Create a `Point` with the coordinates from the oval office on your MapsActivity.
```
Comment (marksdk) [inline src/android/v3/getting-started/directions.md:20]:
```suggestion
Create a `Point` with the coordinates from the Oval Office Location on your MapsActivity.
```

File: src/android/v3/getting-started/directions.md [hunk 4]
```diff
@@ -14,9 +14,374 @@ eleventyNavigation:
...
+    </mi-tab-panel>
+    <mi-tab-panel id="kotlin">
+        <h3>Kotlin</h3>
+        <pre lang ="Kotlin"><code>
+private val mUserLocation: Point = Point(38.897389429704695, -77.03740973527613, 0.0)
+        </code></pre>
+    </mi-tab-panel>
+</mi-tabs>
+
+Now we will make a method that can generate a route for us with just a location (picked through the search list).
```
Comment (marksdk) [inline src/android/v3/getting-started/directions.md:39]:
```suggestion
Now we will make a method that can generate a route for us with just a Location (picked from the search list).
```

File: src/android/v3/getting-started/directions.md [hunk 5]
```diff
@@ -14,9 +14,374 @@ eleventyNavigation:
...
+    </mi-tab-panel>
+    <mi-tab-panel id="kotlin">
+        <h3>Kotlin</h3>
+        <pre lang ="Kotlin"><code>
+class MapsActivity : FragmentActivity(), OnMapReadyCallback, OnRouteResultListener
+        </code></pre>
+    </mi-tab-panel>
+</mi-tabs>
+
+implement the `onRouteResult` method and make the method to generate a route. To generate a route you use `MPRoutingProvider` to query a route with 2 points. We already have a hardcoded starting location so all we need is a point from the location we picked from the search list.
```
Comment (marksdk) [inline src/android/v3/getting-started/directions.md:59]: @timMikkelsen This paragraph is a bit hard to follow. Could try looking at it again?

File: src/android/v3/getting-started/directions.md [hunk 6]
```diff
@@ -14,9 +14,374 @@ eleventyNavigation:
...
+
+So we start by creating an onClickListener on our search `ViewHolder` inside the `SearchItemAdapter` on `onBindViewHolder` that calls a `createRoute` on our activity, to generate a route with the `MPLocation`.
```
Comment (marksdk) [inline src/android/v3/getting-started/directions.md:61]:
```suggestion
To generate a route with the `MPLocation`, we start by creating an `onClickListener` on our search `ViewHolder` inside the `SearchItemAdapter` on `onBindViewHolder`. This calls a `createRoute` on our activity and the route is generated.
```

Comment (marksdk) [reply src/android/v3/getting-started/directions.md:61]: Is the above suggestion still correct code-wise?

File: src/android/v3/getting-started/directions.md [hunk 7]
```diff
@@ -14,9 +14,374 @@ eleventyNavigation:
...
+        <pre lang ="Kotlin"><code>
+holder.itemView.setOnClickListener {
+  mMapActivity.createRoute(mLocations.get(position))
+}
+        </code></pre>
+    </mi-tab-panel>
+</mi-tabs>
+
+When we receive a result on our listener, we render the route through the `MPDirectionsRenderer`.
+We create global variables of the `MPdirectionsRenderer` and `MPRoutingProvider` and create getter to the `MPDirectionsRenderer`to access it from fragments later on.
```
Comment (marksdk) [inline src/android/v3/getting-started/directions.md:85]:
```suggestion

We create global variables of the `MPdirectionsRenderer` and `MPRoutingProvider` and create a getter to the `MPDirectionsRenderer` to access it from _fragments_ later on.
```

File: src/android/v3/getting-started/directions.md [hunk 8]
```diff
@@ -14,9 +14,374 @@ eleventyNavigation:
...
+void createRoute(MPLocation mpLocation) {
+    if (mpRoutingProvider == null) {
+        mpRoutingProvider = new MPRoutingProvider();
+      mpRoutingProvider.setOnRouteResultListener(this);
+    }
+    mpRoutingProvider.query(mUserLocation, mpLocation.getPoint());
+}
+/**
+  * The result callback from the route query. Starts the rendering of the route and opens up a new instance of the navigation fragment on the bottom sheet.
+  * @param route the route model used to render a navigation view.
```
Comment (marksdk) [inline src/android/v3/getting-started/directions.md:102]:
```suggestion
  * `@param route` the route model used to render a navigation view.
```

File: src/android/v3/getting-started/directions.md [hunk 9]
```diff
@@ -14,9 +14,374 @@ eleventyNavigation:
...
+  * @param miError an MIError if anything goes wrong when generating a route
```
Comment (marksdk) [inline src/android/v3/getting-started/directions.md:103]:
```suggestion
  * `@param miError` an `MIError` if anything goes wrong when generating a route
```

Comment (marksdk) [reply src/android/v3/getting-started/directions.md:103]: Should `miError` and `MIError` be the same casing?

File: src/android/v3/getting-started/directions.md [hunk 10]
```diff
@@ -14,9 +14,374 @@ eleventyNavigation:
...
+    </mi-tab-panel>
+    <mi-tab-panel id="kotlin">
+        <h3>Kotlin</h3>
+        <pre lang ="Kotlin"><code>
+class MapsActivity : FragmentActivity(), OnMapReadyCallback, OnRouteResultListener
+        </code></pre>
+    </mi-tab-panel>
+</mi-tabs>
+
+Now we will create a fragment we can put into our BottomSheet to show the steps for each route and the time and distance it takes to travel the route.
```
Comment (marksdk) [inline src/android/v3/getting-started/directions.md:148]:
```suggestion
Now we will create a fragment we can put into our BottomSheet and show the steps for each route, as well as the time and distance it takes to travel the route.
```

File: src/android/v3/getting-started/directions.md [hunk 11]
```diff
@@ -14,9 +14,374 @@ eleventyNavigation:
...
+Here we will use a viewpager to allow the user to switch between each step and a close button. To be able to remove the route and the bottomsheet from the activity.
```
Comment (marksdk) [inline src/android/v3/getting-started/directions.md:149]:
```suggestion

Here we'll use a `viewpager` to allow the user to switch between each step, as well as display a "close" button so we are able to remove the route and the bottom sheet from the activity.
```

File: src/android/v3/getting-started/directions.md [hunk 12]
```diff
@@ -14,9 +14,374 @@ eleventyNavigation:
...
+        android:id="@+id/from_text_view"
+        android:layout_margin="10dp"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_centerInParent="true"
+        android:textStyle="bold"/>
+</RelativeLayout>
+```
+
+Create the Navigation `Fragment` with a `FragmentStateAdapter` for the `ViewPager`
```
Comment (marksdk) [inline src/android/v3/getting-started/directions.md:226]:
```suggestion
Create the Navigation `Fragment` with a `FragmentStateAdapter` for the `ViewPager`:
```

File: src/android/v3/getting-started/directions.md [hunk 13]
```diff
@@ -14,9 +14,374 @@ eleventyNavigation:
...
+    </mi-tab-panel>
+    <mi-tab-panel id="kotlin">
+        <h3>Kotlin</h3>
+        <pre lang ="Kotlin"><code>
+private val mUserLocation: Point = Point(38.897389429704695, -77.03740973527613, 0.0)
+        </code></pre>
+    </mi-tab-panel>
+</mi-tabs>
+
+Create the `RouteLegFragment` for the `ViewPager`
```
Comment (marksdk) [inline src/android/v3/getting-started/directions.md:318]:
```suggestion
Create the `RouteLegFragment` for the `ViewPager`:
```

File: src/android/v3/getting-started/directions.md [hunk 14]
```diff
@@ -14,9 +14,374 @@ eleventyNavigation:
...
+private val mUserLocation: Point = Point(38.897389429704695, -77.03740973527613, 0.0)
+        </code></pre>
+    </mi-tab-panel>
+</mi-tabs>
 
 <!-- Travel-mode -->
 {% include "src/shared/getting-started/directions/travel-mode.md" %}
 
-* **TODO: How to swap between travel modes**
+To swap travel modes you just set the travel mode before making a query for the route.
```
Comment (marksdk) [inline src/android/v3/getting-started/directions.md:370]:
```suggestion
To swap Travel Modes you set the Travel Mode before making a query for the route:
```

File: src/android/v3/getting-started/map.md [hunk 15]
```diff
@@ -14,4 +14,117 @@ eleventyNavigation:
...
+
+## Setup a Google Map with MapsIndoors
+
+### Set the MapsIndoors API Key
+
+In order to include MapsIndoors in your app, you need an API key. If you are not a customer you can use this API key `79f8e7daff76489dace4f9f9` to follow the guide.
+
+In order to include MapsIndoors in your own app with your own content, you need to [contact MapsPeople](https://resources.mapspeople.com/contact-us) to get your building drawings processed and hosted by us.
+
+You will receive a unique API key to use when access has been granted. If you are exploring how this service can become part of your own product, you can read about [partnering with MapsPeople here](https://www.mapspeople.com/become-a-partner).
```
Comment (marksdk) [inline src/android/v3/getting-started/map.md:39]: @timMikkelsen Aren't these instructions in the shared files @simonlaustsen's made?

File: src/android/v3/getting-started/map.md [hunk 16]
```diff
@@ -14,4 +14,117 @@ eleventyNavigation:
...
+
+### Initialize MapsIndoors
+
+Place the following initialization code in the `onCreate` method in the activity that should display the Google map also assign the mapFragment view to a variable as we will use this later to initialize `MapControl`:
```
Comment (marksdk) [inline src/android/v3/getting-started/map.md:43]:
```suggestion
Place the following initialization code in the `onCreate` method in the activity that should display the Google map. You should also assign the `mapFragment` view to a variable as we will use this later to initialize `MapControl`:
```

File: src/android/v3/getting-started/map.md [hunk 17]
```diff
@@ -14,4 +14,117 @@ eleventyNavigation:
...
+    <mi-tab-panel id="kotlin">
+        <h3>Kotlin</h3>
+        <pre lang ="Kotlin"><code>
+MapsIndoors.initialize(applicationContext, "YOUR_MAPSINDOORS_API_KEY")
+MapsIndoors.setGoogleAPIKey(“YOUR_GOOGLE_API_KEY”)
+        </code></pre>
+    </mi-tab-panel>
+</mi-tabs>
+
+In your `onMapReady` callback function, use the `MapControl` class to set up a Google map with MapsIndoors venues, buildings & locations:
```
Comment (marksdk) [inline src/android/v3/getting-started/map.md:66]:
```suggestion
In your `onMapReady` callback function, use the `MapControl` class to set up a Google map with MapsIndoors Venues, Buildings and Locations:
```

File: src/android/v3/getting-started/new-project.md [hunk 18]
```diff
@@ -14,5 +14,73 @@ eleventyNavigation:
...
+
+* Android Development
+* Google Maps Android API
+
+You can get started in two ways, either by reviewing and modifying the [basic example](#basic-example) or do the [clean setup](#setup-mapsindoors).
+
+## Basic Example
+
+You will find in the **Create a link to a very basic implementation of a single activity mapsindoors implementation** a repo that contains the minimum code to start a MapsIndoors project.
+You can also follow the steps below to start your app from scratch or to enhance the Basic Examples, more features will be explained in the [guides](/android/v3/guides/).
```
Comment (marksdk) [inline src/android/v3/getting-started/new-project.md:29]:
```suggestion

You can also follow the steps below to start your app from scratch or to enhance the Basic Examples, more features will be explained in the [guides](/android/v3/guides/).
```

File: src/android/v3/getting-started/new-project.md [hunk 19]
```diff
@@ -14,5 +14,73 @@ eleventyNavigation:
...
+
+## Setup MapsIndoors
+
+If you dont already have a project. We recommend using the Google Maps Activity preset from android studio to getting started on developing your MapsIndoors solution.
```
Comment (marksdk) [inline src/android/v3/getting-started/new-project.md:33]:
```suggestion
If you don't already have a project, we recommend using the Google Maps Activity preset from Android Studio to getting started on developing your MapsIndoors project.
```

File: src/android/v3/getting-started/new-project.md [hunk 20]
```diff
@@ -14,5 +14,73 @@ eleventyNavigation:
...
+
+Add the MapsIndoors SDK as a dependency to your project. The AAR for the MapsIndoors SDK contains both Java classes, SDK resources and AndroidManifest.xml template which gets merged into your application's AndroidManifest.xml during build process. Add or merge in the following to your app's build gradle file (usually called build.gradle).
```
Comment (marksdk) [inline src/android/v3/getting-started/new-project.md:35]:
```suggestion
Add the MapsIndoors SDK as a dependency to your project. The _AAR_ for the MapsIndoors SDK contains both Java classes, SDK resources and an `AndroidManifest.xml` template which gets merged into your application's `AndroidManifest.xml` during build process. Add or merge in the following to your app's build gradle file (usually called `build.gradle`).
```

File: src/android/v3/getting-started/new-project.md [hunk 21]
```diff
@@ -14,5 +14,73 @@ eleventyNavigation:
...
+```java
+android {
+    defaultConfig {
+        minSdkVersion 21
+    }
+    ...
+}
+```
+
+MapsIndoors rely on Java 8 features, so you must add the following compile options, also in *android* section of your *build.gradle* file:
```
Comment (marksdk) [inline src/android/v3/getting-started/new-project.md:48]:
```suggestion
MapsIndoors relies on Java 8 features, so you must add the following compile options, also in *android* section of your *build.gradle* file:
```

File: src/android/v3/getting-started/prerequisites.md [hunk 22]
```diff
@@ -16,3 +16,7 @@ eleventyNavigation:
 
 <!-- MapsIndoors API key -->
 {% include "src/shared/getting-started/prerequisites/api-key-mapsindoors.md" %}
+
+## Work with MapsIndoors SDK behind a Firewall
+
+If you need to work with MapsIndoors SDK behind a firewall, you might need to [white-list some IP-adresses](../../ip-whitelisting).
```
Comment (marksdk) [inline src/android/v3/getting-started/prerequisites.md:22]: @timMikkelsen I made changes to this exact snippet in all other files across the repo. Is this a new addition? If so, it needs to align with these commits: https://github.com/MapsIndoors/docs/commit/c3f158cc55bb094d27230a10dfbcd2ac9171f067

File: src/android/v3/getting-started/search.md [hunk 23]
```diff
@@ -14,15 +14,440 @@ eleventyNavigation:
 <!-- Search -->
 {% include "src/shared/getting-started/search/search.md" %}
 
-* **TODO: Necessary steps to do a simple query search (init search controller, SearchFragment class)**
-* **TODO: A link to advanced search guide and reference guide with all the search parameters explained**
+Start by creating a new activity or fragment to facilitate searches on your application. Here we will be using a fragment to search and show search results on and using a bottom sheet to show the results. We also create a search input field on our main map activity for the user to put in search text.
```
Comment (marksdk) [inline src/android/v3/getting-started/search.md:17]:
```suggestion
Start by creating a new activity or fragment to facilitate searches on your application. Here we will be using a fragment to search and show search results on, and using a bottom sheet to display the results. We also create a search input field on our main map activity for the user to input the text they want to search for.

```

File: src/android/v3/getting-started/search.md [hunk 24]
```diff
@@ -14,15 +14,440 @@ eleventyNavigation:
...
+Both UI component implementations can be found in the getting started app sample.
```
Comment (marksdk) [inline src/android/v3/getting-started/search.md:18]: This should be a link

File: src/android/v3/getting-started/search.md [hunk 25]
```diff
@@ -14,15 +14,440 @@ eleventyNavigation:
...
+
+To perform a search you will need to have initiated `MapsIndoors`. This was shown in the previous section of the getting started tutorial how you do this.
```
Comment (marksdk) [inline src/android/v3/getting-started/search.md:20]: This should be a link.

File: src/android/v3/getting-started/search.md [hunk 26]
```diff
@@ -14,15 +14,440 @@ eleventyNavigation:
...
+
+For advanced usage of the search functionality read the Search guide and tutorials connected to it.
```
Comment (marksdk) [inline src/android/v3/getting-started/search.md:22]: This too 🙂

File: src/android/v3/getting-started/search.md [hunk 27]
```diff
@@ -14,15 +14,440 @@ eleventyNavigation:
...
+    val mpFilter = MPFilter.Builder().setTake(30).build()
+    MapsIndoors.getLocationsAsync(mpQuery, mpFilter) { locations, miError ->
+        //Implement UI handling of the search result here
+    }
+}
+        </code></pre>
+    </mi-tab-panel>
+</mi-tabs>
+
+To start by being able to search. We need a text input field where a user can write what he wants to search for.
```
Comment (marksdk) [inline src/android/v3/getting-started/search.md:61]:
```suggestion
To be able to search we need a text input field where a user can write what they want to search for.

```

Never "he", always "they". It reads the same but is better.

File: src/android/v3/getting-started/search.md [hunk 28]
```diff
@@ -14,15 +14,440 @@ eleventyNavigation:
...
+            return fragment
+        }
+    }
+}
+        </code></pre>
+    </mi-tab-panel>
+</mi-tabs>
+
+To Accompany this we'll create a fragment and a BottomSheet to handle the searchFragment.
+Start by creating a Fragment with a view only consisting of a Recycler View.
```
Comment (marksdk) [inline src/android/v3/getting-started/search.md:155]:
```suggestion
To accompany this we'll create a fragment and a `BottomSheet` to handle the `searchFragment`.

Start by creating a fragment with a view-only consisting of a `RecyclerView`.
```

File: src/android/v3/getting-started/search.md [hunk 29]
```diff
@@ -14,15 +14,440 @@ eleventyNavigation:
...
+            fragment.mMapActivity = mapsActivity
+            return fragment
+        }
+    }
+}
+        </code></pre>
+    </mi-tab-panel>
+</mi-tabs>
+
+Create an RecyclerView adapter and the accompanying Viewholder
```
Comment (marksdk) [inline src/android/v3/getting-started/search.md:232]:
```suggestion
Create a `RecyclerView` adapter and the accompanying `Viewholder`:
```

File: src/android/v3/getting-started/search.md [hunk 30]
```diff
@@ -14,15 +14,440 @@ eleventyNavigation:
...
+    init {
+        text = itemView.findViewById(R.id.text)
+        imageView = itemView.findViewById(R.id.location_image)
+    }
+}
+        </code></pre>
+    </mi-tab-panel>
+</mi-tabs>
+
+Implement a BottomSheet to the bottom of your MapsActivity Layout. The root of the view should be a CoordinatorLayout. You can find the full xml layout on **Link to the github sample with the full view**
```
Comment (marksdk) [inline src/android/v3/getting-started/search.md:356]:
```suggestion
Implement a `BottomSheet` to the bottom of your `MapsActivity` layout. The root of the view should be a `CoordinatorLayout`. You can find the full xml layout on **Link to the github sample with the full view**
```

File: src/android/v3/getting-started/search.md [hunk 31]
```diff
@@ -14,15 +14,440 @@ eleventyNavigation:
...
+        android:clickable="true"
+        style="?attr/bottomSheetStyle"
+        android:background="@drawable/card_shape_bottom"
+        app:layout_behavior="com.google.android.material.bottomsheet.BottomSheetBehavior"
+        app:behavior_hideable="true"
+        android:focusable="true">
+    </FrameLayout>
+```
+
+Now add the search fragment to the bottomSheet in the search query method on your MapsActivity.
```
Comment (marksdk) [inline src/android/v3/getting-started/search.md:372]:
```suggestion
Now add the search fragment to the `bottomSheet` in the search query method on your `MapsActivity`.
```

`bottomSheet` or `BottomSheet`? There are both versions in the text.

File: src/android/v3/getting-started/search.md [hunk 32]
```diff
@@ -14,15 +14,440 @@ eleventyNavigation:
...
+}
+        </code></pre>
+    </mi-tab-panel>
+</mi-tabs>
 
 <!-- Filter map -->
 {% include "src/shared/getting-started/search/filter-map.md" %}
 
-* **TODO: How to filter map based on search results/query**
+When getting a search result you might want to only show thoose search results on the map. You can do this through calling `displaySearchResults(List<MPLocation> locations)` on `MapControl`. This method has different parameters to make it easier for you as a developer to fit your exact need in terms of animation and more. This can be read in the JavaDoc of `MapControl`. The standard implementation animates the camera to fit all locations on the map and shows the info window of a location if its a list of only 1 Location. When you are done showing the different search results you can call `clearMap()` on `MapControl`.
```
Comment (marksdk) [inline src/android/v3/getting-started/search.md:434]:
```suggestion
When getting a search result, you might want to only show those search results on the map. You can do this through calling `displaySearchResults(List<MPLocation> locations)` on `MapControl`. This method has different parameters to make it easier for you as a developer to fit your exact need in terms of animation and more. This can be read in the [JavaDoc of `MapControl`](https://instert-link-here.tim). 

The standard implementation animates the camera to fit all Locations on the map and show the info window of a Location, if it's a list of only one Location. 

When you are done showing the search results you can call `clearMap()` on `MapControl`.
```

I hope you find the Easter egg above or we'll have a 404 on our hands 😉

File: src/android/v3/getting-started/search.md [hunk 33]
```diff
@@ -14,15 +14,440 @@ eleventyNavigation:
...
+    </mi-tab-panel>
+    <mi-tab-panel id="kotlin">
+        <h3>Kotlin</h3>
+        <pre lang ="Kotlin"><code>
+mMapControl.displaySearchResults(locationList)
+        </code></pre>
+    </mi-tab-panel>
+</mi-tabs>
+
+The accompanying ui and implementation of this search experience can be found in the getting started app sample. Found on github: TODO, CREATE GITHUB SAMPLE PAGE.
```
Comment (marksdk) [inline src/android/v3/getting-started/search.md:453]:
```suggestion
The accompanying UI and implementation of this search experience can be found in the getting started app sample. Found on github: TODO, CREATE GITHUB SAMPLE PAGE.
```

**Review by marksdk — CHANGES_REQUESTED** [review]

File: src/android/v3/getting-started/directions.md (hunk 8 above)
Comment (timMikkelsen) [reply src/android/v3/getting-started/directions.md:102]: Should we mix markdown and code examples?

File: src/android/v3/getting-started/directions.md (hunk 9 above)
Comment (timMikkelsen) [reply src/android/v3/getting-started/directions.md:103]: the lower case is the parameter name. Code syntax always have parameters starting with lower case.

File: src/android/v3/getting-started/map.md (hunk 15 above)
Comment (timMikkelsen) [reply src/android/v3/getting-started/map.md:39]: There are some android specific mentions inside the text. But i'll take a look if i cant use the shared files Simon made.

File: src/android/v3/getting-started/prerequisites.md (hunk 22 above)
Comment (timMikkelsen) [reply src/android/v3/getting-started/prerequisites.md:22]: This was probably added before i merged in, a newer version of the docs. I'll put in the new stuff!

File: src/android/v3/getting-started/search.md (hunk 31 above)
Comment (timMikkelsen) [reply src/android/v3/getting-started/search.md:372]: Should be upper case.

File: src/android/v3/getting-started/directions.md (hunk 5 above)
Comment (timMikkelsen) [reply src/android/v3/getting-started/directions.md]: Check the new one. I tried explaining it a bit better.

File: src/android/v3/getting-started/directions.md (hunk 8 above)
Comment (marksdk) [reply src/android/v3/getting-started/directions.md:103]: Ah, no—didn't see this was inside a tab

File: src/android/v3/getting-started/directions.md [hunk 34]
```diff
@@ -14,9 +14,376 @@ eleventyNavigation:
...
+        <h3>Kotlin</h3>
+        <pre lang ="Kotlin"><code>
+class MapsActivity : FragmentActivity(), OnMapReadyCallback, OnRouteResultListener
+        </code></pre>
+    </mi-tab-panel>
+</mi-tabs>
+
+implement the `onRouteResult` method and create a method called `createRoute(MPLocation mpLocation)` To generate a route you can use with `MPRoutingProvider` to query a route with our hardcoded `mUserLocation` and a point from the MPLocation we call the method with later on.
+
+To generate a route with the `MPLocation`, we start by creating an `onClickListener` on our search `ViewHolder` inside the `SearchItemAdapter` on `onBindViewHolder`. This calls a `createRoute` on our activity and the route is generated.
```
Comment (marksdk) [inline src/android/v3/getting-started/directions.md:61]:
```suggestion
Implement the `onRouteResult` method, and create a method called `createRoute(MPLocation mpLocation)`. To generate a route use `MPRoutingProvider` with our hardcoded `mUserLocation` and a point from the `MPLocation`.

To generate a route with the `MPLocation`, we start by creating an `onClickListener` on our search `ViewHolder` inside the `SearchItemAdapter` on `onBindViewHolder`. This calls a `createRoute` on our activity and the route is generated.
```

@timMikkelsen Is what I wrote here still correct? It's still a bit hard to follow for me, but might be spot on for the real developers reading it, so take it as a suggestion only 🙂

**Review by marksdk — CHANGES_REQUESTED** [review]

File: src/android/v3/getting-started/directions.md (hunk 34 above)
Comment (timMikkelsen) [reply src/android/v3/getting-started/directions.md:61]: The query bit is derived from actually being what the method on MPRouteProvider is called. Does it make sense if you read the examples below?

File: src/android/v3/getting-started/new-project.md [hunk 35]
```diff
@@ -14,5 +14,74 @@ eleventyNavigation:
...
+```java
+android {
+    defaultConfig {
+        minSdkVersion 21
+    }
+    ...
+}
+```
+
+MapsIndoors relies on Java 8 features, so you must add the following compile options, also in *android* section of your *build.gradle* file:
```
[the remark this answers is not in the document]
Comment (timMikkelsen) [reply src/android/v3/getting-started/new-project.md:49]: Yes, i am pretty sure it is. Our SDK uses a lot of java 8 features.

File: src/android/v3/getting-started/map.md [hunk 36]
```diff
@@ -14,4 +14,112 @@ eleventyNavigation:
 <!-- Set up MapsIndoors -->
 {% include "src/shared/getting-started/map/show-map.md" %}
 
-* **TODO: Initialize MapsIndoors and show a map with MapsIndoors tiles (MapControl, mi-map-googlemaps component)**
+## Setup Google Maps
```
[the remark this answers is not in the document]
Comment (timMikkelsen) [reply src/android/v3/getting-started/map.md:17]: You are correct on both. I'll remove them and update some things for earlier parts in the getting started guide. To tell the user about starting a project with all of the Google Maps parts already taking care of.

File: src/android/v3/getting-started/new-project.md (hunk 35 above)
[the remark this answers is not in the document]
Comment (Daniel-Nielsen) [reply src/android/v3/getting-started/new-project.md:49]: Ok, I was just asking because I seem to recall it has been like this for quite some time...

[the remark this answers is not in the document]
Comment (timMikkelsen) [reply src/android/v3/getting-started/new-project.md:49]: Yearh, a guess would be that most modern applications use it anyway. It's features like lambda's that are removed if you dont use java 8 😬  
You can read about it here if you are interested: https://developer.android.com/studio/write/java8-support

**Review by Daniel-Nielsen — APPROVED** [review]

File: src/android/v3/getting-started/new-project.md [hunk 37]
```diff
@@ -14,5 +14,74 @@ eleventyNavigation:
...
+* Android Development
+* Google Maps Android API
+
+You can get started in two ways, either by reviewing and modifying the [basic example](#basic-example) or do the [clean setup](#setup-mapsindoors).
+
+## Basic Example
+
+You will find in the **Create a link to a very basic implementation of a single activity mapsindoors implementation** a repo that contains the minimum code to start a MapsIndoors project.
+
+You can also follow the steps below to start your app from scratch or to enhance the Basic Examples, more features will be explained in the [guides](/android/v3/guides/).
```
Comment (marksdk) [inline src/android/v3/getting-started/new-project.md:30]:
```suggestion
You can also follow the steps below to start your app from scratch or to enhance the Basic Examples, more features will be explained in the [guides]({{ site.url }}/android/v3/guides/).
```

File: src/android/v3/guides/getting-started-on-directions.md [hunk 38]
```diff
@@ -18,59 +18,63 @@ To create the Routes you start by querying `MPRoutingProvider` with two Location
...
     }
 }
-        </code></pre>
-    </mi-tab-panel>
+```
+
+</mi-tab-panel>
 </mi-tabs>
 
 To get a Route result we must query the `MPRoutingProvider` with two `Point` objects. First, let's assume we have the latitude and longitude of a user. We create a `Point` object from the latitude and longitude, and set the `zIndex` to `0` as the user is outside. We also need to find a Location the user should navigate to. This can be done by using the Marker the user has selected and creating a UI function to generate a Route for the selected Location. In this example, we will just query for a Location through an ID. We will also set the `TravelMode` to "Walking".
```
Comment (marksdk) [inline src/android/v3/guides/getting-started-on-directions.md:49]:
```suggestion
To get a Route result we must query the `MPRoutingProvider` with two `Point` objects. First, let's assume we have the latitude and longitude of a user. We create a `Point` object from the latitude and longitude, and set the `zIndex` to `0` as the user is outside. We also need to find a Location the user should navigate to. This can be done by using the Marker the user has selected, and creating a UI function to generate a Route for the selected Location. In this example, we will query for a Location through an ID. We will also set the `TravelMode` to "Walking".
```

**Review by marksdk — APPROVED** [review]
